src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -1787,6 +1787,24 @@ private bool IsInTypeofExpression(int position)
             return false;
         }
 
+        /// <summary>
+        /// In the case that there is a valid method group to function pointer conversion, there are two separate syntax nodes:
+        ///  1. The method group itself, which is associated with the BoundMethodGroup, and
+        ///  2. The address of operator, which is associated with the BoundConversion
+        /// The BoundConversion has the correct info for the successful resolution of symbols, but the underlying method group
+        /// has the information for the full member list. We therefore need to treat the BoundMethodGroup as the lowest bound
+        /// node here.
+        /// </summary>
+        private BoundExpression AdjustLowestBoundNodeForFunctionPointerConversion(BoundExpression expr)
```

Comment:
Do we end up with BoundConversion and BoundMethodGroup linked to the same syntax node? It feels like conversion should be linked to the "address of" and the method group should be linked to its operand. Am I misunderstanding the scenario?

Reply:
No, it's the exact opposite. We want to pull the symbols out of the `BoundMethodGroup` that underlies the `BoundConversion` when we call `GetSymbolInfo` or `GetMembers` on `&M`. However, the highest _and_ lowest `BoundNode` for `&M` is the `BoundConversion`: the `BoundMethodGroup` that has the full symbol list we need is associated with the `M` syntax. So we need to grab that node and use it for the lowest node. I can expand the example in this comment.

---
In reply to: [430726079](https://github.com/dotnet/roslyn/pull/44436#discussion_r430726079) [](ancestors = 430726079)